DatabaseWorkshop.md

```diff
@@ -0,0 +1,185 @@
+<img src="https://www.ibm.com/cloud-computing/bluemix/sites/default/files/assets/page/catalog-swift.svg" width="120" alt="Kitura Bird">
+</p>
+
+<p align="center">
+<a href= "http://swift-at-ibm-slack.mybluemix.net/"> 
+    <img src="http://swift-at-ibm-slack.mybluemix.net/badge.svg"  alt="Slack"> 
+</a>
+</p>
+
+## Prerequisite
+This is a follow on tutorial to our [ToDoBackend tutorial](https://github.com/IBM/ToDoBackend). Please complete that before proceeding with this tutorial.
+
+## Installing PostgreSQL
+
+Before we start, ensure you have PostgreSQL installed and running. In Terminal, run:
+```
+brew services start postgresql
+```
+
+If it's already running then skip ahead to step 1. If you do not have it installed then run the following from Terminal:
+```
+brew install postgresql
+brew services start postgresql
+```
+
+## Persistance over local storage
+Adding a database connection means your state remains after you close your server down, and depending on where your database is stored, different applications on different machines can access the data (for instance, in a Cloud application).
+
+### Step 1 - Create a Database
+Create a new database for the project called `tododb` using your Terminal:
+```
+createdb tododb
+```
+
+### Step 2 - Updating Package.swift
+Your `Package.swift` needs two new Packages for the PostgreSQL and ORM to work. 
+1. Open the `ToDoServer` > `Package.swift` file
+2. Add the following two lines to the end of the Package.swift's dependencies section:
+```swift
+.package(url: "https://github.com/IBM-Swift/Swift-Kuery-ORM", from: "0.3.1"),
+.package(url: "https://github.com/IBM-Swift/Swift-Kuery-PostgreSQL", from: "1.2.0"),
+```
+3. Declare these new packages in the "Application" target (note the lack of hyphens):
+```swift
+.target(name: "Application", dependencies: ["SwiftKueryPostgreSQL", "SwiftKueryORM", "KituraCORS", "KituraOpenAPI", "Kitura", "CloudEnvironment","SwiftMetrics","Health",]),
+```
+4. We need to fetch the new dependencies and generate a new Xcode Project file. To do so, make sure Xcode is closed and:
+```bash
+cd ~/ToDoBackend/ToDoServer
+swift package generate-xcodeproj
+open ToDoServer.xcodeproj
+```
+
+### Step 3 - Importing into the Application
+1. In Xcode, open `Sources` > `Application` > `Application.swift`
+2. Add imports for the two new libraries below the other import statements at the top of the file:
+```swift
+import SwiftKueryORM
+import SwiftKueryPostgreSQL
+```
+### Step 4 - Extending ToDo to conform to Model
+In order for your `ToDo` struct located in `Application` > `Model.swift`, we need to conform `ToDo` to Swift-Kuery-ORMs Model protocol. At the bottom of `Application.swift` add:
+```swift
+extension ToDo: Model {
+}
+```
+This exposes new methods to ToDo for saving, finding, deleting etc. from a database.
+
+### Step 5 - Creating a connection with the database
+Add the following Class to the bottom of the file, outside the `App` classes final curly brace:
```

Comment:
Add the following code, to set up your database connection pool, to the end of the `Application.swift` file, below the final curly brace:

Reply:
✅